keanu-project/src/main/java/io/improbable/keanu/tensor/dbl/Nd4jDoubleTensor.java

```diff
@@ -719,19 +743,21 @@ public DoubleTensor getLessThanOrEqualToMask(DoubleTensor lessThanOrEqualToThis)
     public DoubleTensor setWithMaskInPlace(DoubleTensor mask, Double value) {
 
         INDArray maskDup = unsafeGetNd4J(mask).dup();
-
+        double trueValue = 1.0;
```

Comment:
How is this change related to this PR?

Reply:
it's a bug fix, and it is related to the rest of the PR: previously `setWithMaskInPlace` didn't work for `NaN`, i.e. it didn't set the tensor in those places where the tensor value was `NaN`.

I've captured this failure case in `Nd4jDoubleTensorTest#canSetToZeroWhenNaN`